### PR TITLE
Cpdrp 701 readd withdrawn participants

### PR DESCRIPTION
**Pull request by ethanmills**
### Context
CPDRP-701
Once participant profiles have been marked as withdrawn, we should allow new profiles to be created for those users

### Changes proposed in this pull request
- Ensure withdrawn profiles are displayed correctly in the participants API
- Ensure withdrawn participants are not displayed to admins
- Allow withdrawn participants to be readded by SITs
- Update email chaser logic to account for withdrawn participants

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?
- sign in as school-leader@example.com and add a participant
- sign in as admin@example.com and remove the particiapant
- sign back in as school-leader and add the same details again

Ensure they are displayed to the school tutor and the admin correctly

### CODE REVIEW

File: app/forms/schools/add_participant_form.rb [hunk 1]
```diff
@@ -79,7 +79,12 @@ def mentor
     end
 
     def email_already_taken?
-      User.find_by(email: email)&.participant_profiles&.ecf&.any?
+      User.find_by(email: email)
+        &.teacher_profile
+        &.participant_profiles
```
Comment (kicferk1) [inline app/forms/schools/add_participant_form.rb:84]: do we want `participant_profiles through teacher_profile` on user model? perhaps not, up for discussion

Comment (ethanmills) [reply app/forms/schools/add_participant_form.rb:84]: Think that's a topic for 2pm 🙂

Comment (kicferk1) [reply app/forms/schools/add_participant_form.rb:84]: Eeeeh, I have a clash 😢 

In general I am not sure how true the code comment in User model
```
# TODO: Legacy associations, to be removed
```
is. There are some parts of our app that want profiles and users to be closely related - in particular Track and Pay apis talk about 'participants' - which are users - and there is quite a bit of logic around validating them that wants to know which profiles that user has. Maybe at some point we will switch them to operate on teacher profiles instead, but, ehh, not sure when that would be?

File: spec/forms/schools/add_participant_form_spec.rb [hunk 2]
```diff
@@ -31,8 +31,8 @@
       it "sets the participant_type to mentor as well as full name and email to match current_user details" do
         expect { subject.type = "self" }
           .to change { subject.participant_type }.to(:mentor)
-          .and change { subject.full_name }.to(user.full_name)
-          .and change { subject.email }.to(user.email)
+                                                 .and change { subject.full_name }.to(user.full_name)
+                                                                                  .and change { subject.email }.to(user.email)
```
Comment (kicferk1) [inline spec/forms/schools/add_participant_form_spec.rb:35]: uff, that's a lot of whitespace :P 

the old spacing looked nice, the new one looks... less nice 😭

File: spec/requests/api/v1/participants_spec.rb [hunk 3]
```diff
@@ -8,13 +8,17 @@
...
     let(:bearer_token) { "Bearer #{token}" }
 
     before :each do
       mentor_profile = create(:mentor_profile, school: partnership.school, cohort: partnership.cohort)
-      create_list :early_career_teacher_profile, 2, mentor_profile: mentor_profile, school: partnership.school, cohort: partnership.cohort
+      create_list :early_career_teacher_profile, 2, mentor_profile: mentor_profile, school_cohort: school_cohort
+      ect_teacher_profile = ParticipantProfile::ECT.first.teacher_profile
+      create(:participant_profile, :withdrawn, :ect, teacher_profile: ect_teacher_profile, school_cohort: school_cohort)
     end
+    let!(:withdrawn_ect_profile) { create(:participant_profile, :withdrawn, :ect, school_cohort: school_cohort) }
```
Comment (kicferk1) [inline spec/requests/api/v1/participants_spec.rb:21]: So we have two withdrawn ect profiles? One for fresh user, one for a user that has another, active participant profile?

Comment (kicferk1) [reply spec/requests/api/v1/participants_spec.rb:21]: I wonder if there is a way to make it more obvious to the reader of the test.

File: app/forms/schools/add_participant_form.rb (hunk 1 above)
Comment (ethanmills) [reply app/forms/schools/add_participant_form.rb:84]: yesterday 2pm never ended up happening. I still think this is a wider conversation than we can have here

File: spec/requests/api/v1/participants_spec.rb (hunk 3 above)
Comment (ethanmills) [reply spec/requests/api/v1/participants_spec.rb:21]: Quite a hard thing to name. will try

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**